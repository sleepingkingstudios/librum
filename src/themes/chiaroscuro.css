@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  .theme-chiaroscuro {
    @apply bg-zinc-100 text-zinc-800;
    @apply dark:bg-zinc-900 dark:text-zinc-200;

    h1, h2, h3, h4, h5, h6 {
      @apply font-serif text-zinc-900;
      @apply dark:text-zinc-100;
    }

    a {
      @apply text-blue-500;
      @apply hover:text-blue-600 hover:underline;
      @apply dark:hover:text-blue-400;
    }

    hr {
      @apply border-zinc-900;
      @apply dark:border-zinc-100;
    }

    .hr-fancy {
      @apply after:border-zinc-900;
      @apply dark:after:border-zinc-100;

      border-top-style: solid;
      border-top-width: 2px;
      opacity: 1;

      &:after {
        content: ' ';

        /* Display */
        display: block;
        border-top-style: solid;
        border-top-width: 1px;
        margin: 0 10px;

        /* Positioning */
        position: relative;
        top: 2px;
      }
    }

    .hr-muted {
      @apply border-zinc-400 after:border-zinc-400;
      @apply dark:border-zinc-600 dark:after:border-zinc-600;
    }
  }
}

@layer components {
  .theme-chiaroscuro {
    .page-navigation {
      .dropdown {
        @apply bg-zinc-100/75;
        @apply dark:bg-zinc-900/75;
      }

      .dropdown-item {
        @apply font-serif;
      }

      .dropdown-item-link {
        @apply text-zinc-900 hover:text-black;
        @apply dark:text-zinc-100 dark:hover:text-white;
      }

      .dropdown-label {
        @apply font-serif text-zinc-900;
        @apply dark:text-zinc-100;
      }

      .navigation-item {
        @apply font-serif;
      }

      .navigation-item-link {
        @apply text-zinc-900 hover:text-black;
        @apply dark:text-zinc-100 dark:hover:text-white;
      }
    }

    .page-subtitle {
      @apply font-serif text-zinc-900;
      @apply dark:text-zinc-100;
    }

    .page-title {
      @apply font-serif text-zinc-900;
      @apply dark:text-zinc-100;
    }
  }
}

@layer utilities {
  .theme-chiaroscuro {
    .link-danger {
      @apply text-red-600;
      @apply hover:text-red-700;
      @apply dark:hover:text-red-500;
    }

    .link-muted {
      @apply text-zinc-500;
      @apply hover:text-zinc-600;
      @apply dark:hover:text-zinc-400;
    }

    .text-danger {
      @apply text-red-600;
    }

    .text-muted {
      @apply text-zinc-500;
    }
  }
}
