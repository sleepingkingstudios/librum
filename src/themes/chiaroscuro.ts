import {
  Theme,
  defaultTheme,
} from './theme';

export const chiaroscuro: Theme = {
  ...defaultTheme,
  background: 'bg-chiaroscuro-background dark:bg-chiaroscuro-background-dark',
  header: '@headerBase text-chiaroscuro-text-strong dark:text-chiaroscuro-text-strong-dark',
  headerBase: 'font-serif',
  hr: 'border-chiaroscuro-text-strong dark:border-chiaroscuro-text-strong-dark after:border-chiaroscuro-text-strong after:dark:border-chiaroscuro-text-strong-dark',
  hrMuted: 'border-chiaroscuro-text-muted after:dark:border-chiaroscuro-text-muted',
  link: '@linkBase text-chiaroscuro-link hover:text-chiaroscuro-link-hover dark:hover:text-chiaroscuro-link-hover-dark',
  linkBase: 'hover:underline',
  linkDanger: '@linkBase text-chiaroscuro-text-danger hover:text-chiaroscuro-text-danger-hover dark:text-chiaroscuro-text-danger-dark dark:hover:text-chiaroscuro-text-danger-hover-dark',
  name: 'Chiaroscuro',
  text: '@textBase text-chiaroscuro-text dark:text-chiaroscuro-text-dark',
  textBase: 'font-sans',
  textDanger: '@textBase text-chiaroscuro-text-danger dark:text-chiaroscuro-text-danger-dark',
  textMuted: '@textBase text-chiaroscuro-text-muted',
};
