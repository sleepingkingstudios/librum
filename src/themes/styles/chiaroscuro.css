@tailwind base;
@tailwind components;
@tailwind utilities;

.theme-chiaroscuro {
  --color-bg: #fafafa; /* zinc-100 */
  --color-bg-dark: #18181b; /* zinc-900 */
  --color-bg-accent: #e4e4e7; /* zinc-200 */
  --color-bg-accent-dark: #27272a; /* zinc-800 */

  --color-text: var(--color-bg-dark-accent);
  --color-text-dark: var(--color-bg-accent);
  --color-text-accent: var(--color-bg-dark);
  --color-text-accent-dark: var(--color-bg);

  /* ACCENT COLORS */

  --color-danger: #dc2626; /* red-600 */
  --color-danger-dark: var(--color-danger);
  --color-danger-accent: #b91c1c; /* red-700 */
  --color-danger-accent-dark: #ef4444; /* red-500 */
  --color-danger-bg: #fee2e2; /* red-100 */
  --color-danger-bg-dark: #3f0808; /* red-950 ish */

  --color-info: #22d3ee; /* cyan-400 */
  --color-info-dark: #06b6d4; /* cyan-500 */
  --color-info-accent: #06b6d4; /* cyan-500 */
  --color-info-accent-dark: #22d3ee; /* cyan-400 */
  --color-info-bg: #ecfeff; /* cyan-50 */
  --color-info-bg-dark: #0b2731; /* cyan-950 ish */

  --color-muted: #a1a1aa; /* zinc-400 */
  --color-muted-dark: #52525b; /* zinc-600 */
  --color-muted-accent: #71717a; /* zinc-500 */
  --color-muted-accent-dark: #71717a; /* zinc-500 */
  --color-muted-bg: #f4f4f5; /* zinc-100 */
  --color-muted-bg-dark: #27272a; /* zinc-800 */

  --color-primary: #0284c7; /* sky-600 */
  --color-primary-dark: var(--color-primary);
  --color-primary-accent: #0369a1; /* sky-700 */
  --color-primary-accent-dark: #0ea5e9; /* sky-500 */
  --color-primary-bg: #e0f2fe; /* sky-100 */
  --color-primary-bg-dark: #062537; /* sky-950 ish */

  --color-success: #10b981; /* emerald-500 */
  --color-success-dark: #059669; /* emerald-600 */
  --color-success-accent: #059669; /* emerald-600 */
  --color-success-accent-dark: #10b981; /* emerald-500 */
  --color-success-bg: #ecfdf5; /* emerald-50 */
  --color-success-bg-dark: #03271d; /* emerald-950 ish */

  --color-warning: #f59e0b; /* amber-500 */
  --color-warning-dark: var(--color-warning);
  --color-warning-accent: #d97706; /* amber-600 */
  --color-warning-accent-dark: #fbbf24; /* amber-400 */
  --color-warning-bg: #fffbeb; /* amber-50 */
  --color-warning-bg-dark: #1e0d03; /* amber-975 ish */
}

@layer base {
  .theme-default-styles.theme-chiaroscuro {
    /* LEGACY COLORS */

    --bg-color: #fafafa; /* zinc-100 */
    --bg-color-dark: #18181b; /* zinc-900 */
    --bg-color-accent: #e4e4e7; /* zinc-200 */
    --bg-color-accent-dark: #27272a; /* zinc-800 */

    --link-color: #3b82f6; /* blue-500 */
    --link-color-hover: #2563eb; /* blue-600 */
    --link-color-hover-dark: #60a5fa; /* blue-400 */

    --muted-color: #a1a1aa; /* zinc-400 */
    --muted-color-dark: #52525b; /* zinc-600 */

    --text-color: var(--bg-color-accent-dark);
    --text-color-dark: var(--bg-color-accent);
    --text-color-strong: var(--bg-color-dark);
    --text-color-strong-dark: var(--bg-color);

    @apply dark:bg-[color:var(--bg-color-dark)] dark:text-[color:var(--text-color-dark)];

    h1, h2, h3, h4, h5, h6 {
      @apply dark:text-[color:var(--text-color-strong-dark)];
    }

    a {
      @apply dark:hover:text-[color:var(--link-color-hover-dark)];
    }

    hr {
      @apply border-[color:var(--bg-color-dark)] after:border-[color:var(--bg-color-dark)];
      @apply dark:border-[color:var(--bg-color)] dark:after:border-[color:var(--bg-color)];
    }
  }
}

@layer components {
  .theme-default-styles.theme-chiaroscuro {
    .button, .form-button {
      @apply border-zinc-800 text-zinc-800;
      @apply active:bg-zinc-200 hover:border-zinc-900 hover:text-zinc-900;
      @apply dark:border-zinc-200 dark:text-zinc-200;
      @apply dark:active:bg-zinc-800 dark:hover:border-zinc-100 dark:hover:text-zinc-100;
    }

    .button:disabled, .form-button:disabled {
      @apply border-zinc-500 text-zinc-500;
      @apply active:bg-zinc-100 dark:active:bg-zinc-900;
    }

    .data-table {
      tbody tr:nth-child(odd) {;
        @apply bg-[color:var(--bg-color-accent)];
        @apply dark:bg-[color:var(--bg-color-accent-dark)];
      }
    }

    .form-field-errors {
      @apply text-red-500;
    }

    .form-label {
      @apply text-zinc-900;
      @apply dark:text-zinc-100;
    }

    .form-loading-overlay {
      @apply bg-zinc-100/75;
      @apply dark:bg-zinc-900/75;
    }

    .form-input {
      @apply bg-zinc-100 dark:bg-zinc-900 border-zinc-500;
    }

    .page-breadcrumbs {
      .page-breadcrumb {
        @apply sm:before:text-zinc-500;

        .page-breadcrumb-empty {
          @apply text-zinc-500;
        }
      }
    }

    .page-navigation {
      .dropdown {
        @apply bg-zinc-100/75;
        @apply dark:bg-zinc-900/75;
      }

      .dropdown-item-link {
        @apply text-zinc-900 hover:text-black;
        @apply dark:text-zinc-100 dark:hover:text-white;
      }

      .dropdown-label {
        @apply text-zinc-900;
        @apply dark:text-zinc-100;
      }

      .navigation-item-link {
        @apply text-zinc-900 hover:text-black;
        @apply dark:text-zinc-100 dark:hover:text-white;
      }
    }

    .page-subtitle {
      @apply text-zinc-900;
      @apply dark:text-zinc-100;
    }

    .page-title {
      @apply text-zinc-900;
      @apply dark:text-zinc-100;
    }
  }
}

@layer utilities {
  .theme-default-styles.theme-chiaroscuro {
    .alert-failure {
      @apply bg-red-500 border-red-600;
      @apply dark:bg-red-600 dark:border-red-700;
    }

    .alert-info {
      @apply bg-blue-300 border-blue-400;
      @apply dark:bg-blue-700 dark:border-blue-600;
    }

    .alert-success {
      @apply bg-emerald-500 border-emerald-600;
      @apply dark:bg-emerald-700 dark:border-emerald-600;
    }

    .alert-warning {
      @apply bg-amber-500 border-amber-600;
      @apply dark:bg-amber-600 dark:border-amber-500;
    }

    .button-info {
      @apply border-blue-600 text-blue-600;
      @apply active:bg-blue-200 hover:border-blue-700 hover:text-blue-700;
      @apply dark:border-blue-400 dark:text-blue-400;
      @apply dark:active:bg-blue-900 dark:hover:border-blue-300 dark:hover:text-blue-300;
    }

    .hr-fancy {
      @apply after:border-zinc-900;
      @apply dark:after:border-zinc-100;
    }

    .hr-muted {
      @apply border-zinc-400 after:border-zinc-400;
      @apply dark:border-zinc-600 dark:after:border-zinc-600;
    }

    .link-danger {
      @apply text-red-600;
      @apply hover:text-red-700;
      @apply dark:hover:text-red-500;
    }

    .link-muted {
      @apply text-zinc-500;
      @apply hover:text-zinc-600;
      @apply dark:hover:text-zinc-400;
    }

    .link-plain {
      @apply text-zinc-900 hover:text-black;
      @apply dark:text-zinc-100 dark:hover:text-white;
    }
  }
}
