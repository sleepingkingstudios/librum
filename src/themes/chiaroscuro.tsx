import * as React from 'react';

import type { Theme } from './types';

import './styles/chiaroscuro.css';
import './styles/default-styles.css';

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const ChiaroscuroTheme = () => (<span className="theme-chiaroscuro" />);

export const chiaroscuro: Theme = {
  className: 'theme-chiaroscuro theme-default-styles',
  name: 'chiaroscuro',
};
