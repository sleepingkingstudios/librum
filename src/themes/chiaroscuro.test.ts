import { chiaroscuro } from './chiaroscuro';

describe('chiaroscuro Theme', () => {
  it('should match the snapshot', () => {
    expect(chiaroscuro).toMatchSnapshot();
  });
});
