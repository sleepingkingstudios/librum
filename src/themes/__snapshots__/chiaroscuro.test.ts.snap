// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chiaroscuro Theme should match the snapshot 1`] = `
Object {
  "background": "bg-chiaroscuro-background dark:bg-chiaroscuro-background-dark",
  "header": "@headerBase text-chiaroscuro-text-strong dark:text-chiaroscuro-text-strong-dark",
  "headerBase": "font-serif",
  "headerDanger": "@headerBase text-chiaroscuro-text-danger-hover dark:text-chiaroscuro-text-danger-hover-dark",
  "hr": "border-chiaroscuro-text-strong dark:border-chiaroscuro-text-strong-dark after:border-chiaroscuro-text-strong after:dark:border-chiaroscuro-text-strong-dark",
  "hrMuted": "border-chiaroscuro-text-muted after:border-chiaroscuro-text-muted",
  "link": "@linkBase text-chiaroscuro-link hover:text-chiaroscuro-link-hover dark:hover:text-chiaroscuro-link-hover-dark",
  "linkBase": "hover:underline",
  "linkDanger": "@linkBase text-chiaroscuro-text-danger hover:text-chiaroscuro-text-danger-hover dark:text-chiaroscuro-text-danger-dark dark:hover:text-chiaroscuro-text-danger-hover-dark",
  "name": "Chiaroscuro",
  "subtitle": "@headerBase text-2xl",
  "text": "@textBase text-chiaroscuro-text dark:text-chiaroscuro-text-dark",
  "textAccent": "@textBase",
  "textBase": "font-sans",
  "textDanger": "@textBase text-chiaroscuro-text-danger dark:text-chiaroscuro-text-danger-dark",
  "textHeading": "@textBase",
  "textMuted": "@textBase text-chiaroscuro-text-muted",
  "textStrong": "@textBase",
  "title": "@headerBase text-5xl",
}
`;
